### ISSUE TITLE
Question: What Kubernetes versions should we support?

### ISSUE THREAD

**Issue by clockworksoul** [clockworksoul/helm-elasticsearch#31]
Hello all. We should probably decide what versions of Kubernetes we officially support.

I'm personally leaning towards 1.6 through 1.8, unless there's a sufficient number of 1.5 clusters in use to justify support of version 1.5?

**Comment by mikn** [comment]
I'd personally follow the official charts guidelines to only support the most recent two versions, so 1.7 and 1.8, but maybe that's a bit harsh?

**Comment by jjungnickel** [comment]
@mikn I think it's sensible to follow the official guidelines, providing a consistent expectation baseline. Old chart releases supporting older cluster versions don't have to go away. But development should focus on this subset.

**Comment by sagikazarmark** [comment]
As long as maintaining support for older versions does not cost us anything I would say we shouldn't automatically drop support (eg. drop old syntax if forward compatible) for older versions, but I think we can communicate this as the guideline (last two versions are supported).

**Comment by clockworksoul** [comment]
> As long as maintaining support for older versions does not cost us anything I would say we shouldn't automatically drop support (eg. drop old syntax if forward compatible) for older versions, but I think we can communicate this as the guideline (last two versions are supported).

This implies that we shouldn't be reluctant to remove support for earlier versions if it gets in the way of adding support for a valuable addition for a newer version. I absolutely agree with this.

Similarly, adding features that _require_ the newest version are fine, so long as the chart is smart enough to exclude the feature if it's being installed on an older (but still supported) version cluster.

> I'd personally follow the official charts guidelines to only support the most recent two versions, so 1.7 and 1.8, but maybe that's a bit harsh?

That does seem a little harsh. I'd personally like to try to support the current and previous two versions (currently 1.7, 1.8, and 1.9). I know from experience that there's still quite a few people out there running 1.7 (even 1.6!) in production.

**Comment by sagikazarmark** [comment]
> I know from experience that there's still quite a few people out there running 1.7 (even 1.6!) in production.

Agreed, not everyone has the resources to always keep up with the latest greatest version on production systems.
